relay-server/src/services/processor/dynamic_sampling.rs

```diff
@@ -140,10 +137,10 @@ pub fn sample_envelope_items(
     }
 }
 
-/// Computes the sampling decision on the incoming transaction.
+/// Computes the sampling decision on the incoming envelope.
 fn compute_sampling_decision(
     processing_enabled: bool,
-    reservoir: &ReservoirEvaluator,
+    reservoir: Option<&ReservoirEvaluator>,
```

Comment:
I really don't like this `Option<&ReservoirEvaluator>` everywhere, can't we have a `ReservoirEvaluator::disabled()` instead or make it a trait and `Option<ReservoirEvaluator>` implements the trait.

Maybe a future refactor.